mono/metadata/image.c

```diff
@@ -2578,10 +2578,10 @@ mono_find_mempool_owner (void *ptr)
 	gboolean searching = TRUE;
 
 	// Iterate over both by-path image hashes
-	const int hash_count = 2;
-	const int hash_candidates[hash_count] = {IMAGES_HASH_PATH, IMAGES_HASH_PATH_REFONLY};
+	#define HASH_CANDIDATE_COUNT 2
+	const int hash_candidates[HASH_CANDIDATE_COUNT] = {IMAGES_HASH_PATH, IMAGES_HASH_PATH_REFONLY};
 	int hash_idx;
-	for (hash_idx = 0; searching && hash_idx < IMAGES_HASH_COUNT; hash_idx++)
+	for (hash_idx = 0; searching && hash_idx < HASH_CANDIDATE_COUNT; hash_idx++)
```

Comment:
Not familiar with the code, but this looks to be changing semantics while the commit itself just makes it sound like a build fix.